lib/scenario.js

```diff
@@ -81,9 +81,8 @@ module.exports.injected = function (fn, suite, hookName) {
     }
     recorder.catch((err) => {
       event.emit(event.test.failed, suite, err); // emit
-      if (hookName === 'after' || hookName === 'afterSuite') {
-        event.emit(event.hook[hookName].failed, suite);
-      }
+      if (hookName === 'after') event.emit(event.test.after, test);
```

Comment:
Also there is an issue in bdd.js
now
context.After = function (fn) {
      afterEachHooks.unshift(['After', scenario.injected(fn, suites[0]), 'after']);
    };

should be
context.After = function (fn) {
      afterEachHooks.unshift(['After', scenario.injected(fn, suites[0], 'after')]);
    };

Because of this we don't get name of hook in this place for After